Run gitlab CI build jobs on Fedora 41

achilleas-k opened the pull request:
**Schutzfile: switch Fedora 41 repos to released**


---

**Schutzfile: define common gitlab-ci-runner**


---

**imgtestlib: read common CI runner from Schutzfile**

Use the common CI runner defined in the Schutzfile in all gitlab CI
generators.

---

**gitlab: bump gitlab CI runners to Fedora 41**


---

**schutzbot: update terraform commit ID**


---

thozza commented on test/scripts/imgtestlib.py:
```diff
@@ -18,6 +18,11 @@
+# CI runner for common tasks.
+# Currently this is used for all gitlab CI jobs. In the future, we might switch to running build jobs on the same host
+# distro as the target image, but this CI runner will still be used for generic tasks like check-build-coverage.
+CI_RUNNER = "aws/fedora-40"
```
A bit of a nitpick, but it would be IMHO nicer to have the `CI_RUNNER` defined in a configuration file, such as `Schutzfile`, instead of in a script.

achilleas-k replied:
I considered it for a bit but was on the fence.  Since you brought it up, I'll do it.

thozza reviewed (approved):
Thanks for doing the :broom: 

I've added one idea, but don't consider it a blocker.

achilleas-k commented:
Added the CI runner definition to the `Schutzfile` and updated the terraform SHA to the newest which includes Fedora 41 runner definitions.

thozza reviewed (approved):
Amazing! :heart_eyes:

achilleas-k commented:
A lot of jobs timed out in GitLab.  I'm hoping it's because this PR was running at the same time as #1066, both of which caused a rebuild of every image.

I hit the rebase button here now, which means a full rebuild again, but it also means we wont need one on the merge queue.

thozza commented:
@achilleas-k try using the latest `main` HEAD from terraform repo. I updated the F41 AMIs https://github.com/osbuild/gitlab-ci-terraform/pull/176

achilleas-k commented:
> @achilleas-k try using the latest `main` HEAD from terraform repo. I updated the F41 AMIs [osbuild/gitlab-ci-terraform#176](https://github.com/osbuild/gitlab-ci-terraform/pull/176)

Rebased, resolved conflicts, and updated the commit ID.

Thanks!!
